explainability-service/src/main/java/org/kie/trustyai/service/endpoints/explainers/local/LimeEndpoint.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2023 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kie.trustyai.service.endpoints.explainers.local;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import javax.enterprise.inject.Instance;
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+
+import org.eclipse.microprofile.openapi.annotations.tags.Tag;
+import org.jboss.logging.Logger;
+import org.kie.trustyai.explainability.local.lime.LimeConfig;
+import org.kie.trustyai.explainability.local.lime.LimeExplainer;
+import org.kie.trustyai.explainability.model.Prediction;
+import org.kie.trustyai.explainability.model.PredictionInput;
+import org.kie.trustyai.explainability.model.PredictionInputsDataDistribution;
+import org.kie.trustyai.explainability.model.PredictionProvider;
+import org.kie.trustyai.service.config.ServiceConfig;
+import org.kie.trustyai.service.data.DataSource;
+import org.kie.trustyai.service.payloads.BaseExplanationResponse;
+import org.kie.trustyai.service.payloads.LocalExplanationRequest;
+import org.kie.trustyai.service.payloads.SaliencyExplanationResponse;
+
+@Tag(name = "LIME Explainer Endpoint")
+@Path("/explainers/local/lime")
+public class LimeEndpoint extends LocalExplainerEndpoint {
+
+    private static final Logger LOG = Logger.getLogger(LimeEndpoint.class);
+    @Inject
+    Instance<DataSource> dataSource;
+
+    @Inject
+    ServiceConfig serviceConfig;
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response explain(LocalExplanationRequest request) {
+        return processRequest(request, dataSource.get(), serviceConfig);
+    }
+
+    @Override
+    public BaseExplanationResponse generateExplanation(PredictionProvider model, Prediction predictionToExplain, List<PredictionInput> inputs) {
+        LimeConfig config = new LimeConfig().withDataDistribution(new PredictionInputsDataDistribution(inputs));
+        LimeExplainer limeExplainer = new LimeExplainer(config); //TODO: switch to RecordingLimeExplainer?
```

Comment:
TODO to be fixed? Future ticket?

Reply:
yeah, I'll create a new ticket for this.